app/src/main/java/org/schabi/newpipe/NewVersionWorker.kt

```diff
@@ -0,0 +1,122 @@
+package org.schabi.newpipe
+
+import android.content.Context
+import android.util.Log
+import androidx.core.content.edit
+import androidx.preference.PreferenceManager
+import androidx.work.OneTimeWorkRequest
+import androidx.work.WorkManager
+import androidx.work.WorkRequest
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import com.grack.nanojson.JsonParser
+import com.grack.nanojson.JsonParserException
+import org.schabi.newpipe.extractor.downloader.Response
+import org.schabi.newpipe.extractor.exceptions.ReCaptchaException
+import org.schabi.newpipe.util.ReleaseUtil.coerceExpiry
+import org.schabi.newpipe.util.ReleaseUtil.compareAppVersionAndShowNotification
+import org.schabi.newpipe.util.ReleaseUtil.isExpired
+import org.schabi.newpipe.util.ReleaseUtil.isReleaseApk
+import java.io.IOException
+
+class NewVersionWorker(
+    context: Context,
+    workerParams: WorkerParameters
+) : Worker(context, workerParams) {
+
+    @Throws(IOException::class, ReCaptchaException::class)
+    private fun checkNewVersion() {
+        // Check if the current apk is a github one or not.
+        if (!isReleaseApk()) {
+            return
+        }
+
+        val prefs = PreferenceManager.getDefaultSharedPreferences(applicationContext)
+        // Check if the last request has happened a certain time ago
+        // to reduce the number of API requests.
+        val expiry = prefs.getLong(applicationContext.getString(R.string.update_expiry_key), 0)
+        if (!isExpired(expiry)) {
+            return
+        }
```

Comment:
Can the expiration date be handled by some parameter in the worker?

Reply:
I mean I guess, but I'm not sure why? Could you expand on this?

Reply:
Nevermind, there [doesn't seem](https://developer.android.com/reference/androidx/work/WorkRequest.Builder) [to be](https://developer.android.com/reference/androidx/work/Constraints) any method that allows scheduling a work only if it has not been already run in the previous X hours. Stick with this, sorry for the confusion.